packages/dapp/src/components/Widgets/WidgetEvents.tsx

```diff
@@ -72,14 +87,35 @@ export function WidgetEvents() {
           lastTxHashRef.current = update.process.txHash;
           trackTransaction({
             chain: update.route.fromChainId,
-            transactionHash: update.process.txHash || '',
-            category: TrackingCategories.WidgetEvent,
-            action: TrackingActions.OnRouteExecutionUpdated,
+            txhash: update.process.txHash || '',
+            category: TrackingCategory.WidgetEvent,
+            action: TrackingAction.OnRouteExecutionUpdated,
             data: {
-              routeId: `${update.route.id}`,
-              transactionLink: update.process.txLink,
-              steps: update.route.steps,
-              status: update.process.status,
+              label: 'execution_update',
+              [TrackingEventParameter.FromAmountUSD]:
+                update.route.fromAmountUSD,
+              [TrackingEventParameter.ToAmountUSD]: update.route.toAmountUSD,
+              [TrackingEventParameter.FromAmount]: update.route.fromAmount,
+              [TrackingEventParameter.ToAmount]: update.route.toAmount,
+              [TrackingEventParameter.FromToken]:
+                update.route.fromToken.address,
+              [TrackingEventParameter.ToToken]: update.route.toToken.address,
+              [TrackingEventParameter.FromChainId]: update.route.fromChainId,
+              [TrackingEventParameter.ToChainId]: update.route.toChainId,
+              [TrackingEventParameter.RouteId]: `${update.route.id}`,
+              [TrackingEventParameter.Status]: update.process.status,
+              [TrackingEventParameter.TxHash]: update.process.txHash || '',
+              [TrackingEventParameter.TxLink]: update.process.txLink || '',
+              [TrackingEventParameter.Type]: update.process.type,
+              [TrackingEventParameter.GasCostUSD]: update.route.gasCostUSD,
+              [TrackingEventParameter.ErrorCode]:
+                update.process.error?.code || '',
+              [TrackingEventParameter.ErrorMessage]:
+                update.process.error?.message || '',
+              [TrackingEventParameter.InsuranceFeeAmountUSD]:
+                update.route.insurance.feeAmountUsd,
+              [TrackingEventParameter.InsuranceState]:
```

Comment:
Data is quite huge, let's use it via variable?

Reply:
@Abhikumar98 

Thanks for sharing that idea! Generally, I am with you on re-structuring bigger chunks into small pieces for the sake of clean and readable code. 
In this case, imo it´s all in that spot where it belongs, easy to read and follow up. I do not see the need of seperation.

@Addminus Wdyt?